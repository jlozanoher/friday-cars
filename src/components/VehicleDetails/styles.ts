import styled from "styled-components";

export const Title = styled.div`
  font-size: 20px;
  font-weight: 500;
  line-height: 24px;
`;

export const SubTitle = styled.div`
  font-size: 16px;
  line-height: 24px;
`;

export const Img = styled.img`
  width: 250px;
  margin: 16px 8px;
`;

export const SelectBtn = styled.div`
  padding: 8px;
  background: #1b875a;
  border-radius: 4px;
  font-weight: 400;
  cursor: pointer;

  &.red {
    background: #b54124;
  }
`;
