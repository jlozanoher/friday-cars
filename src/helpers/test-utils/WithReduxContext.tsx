import { Provider } from "react-redux";
import store from "../../app/store";

interface Props {
  children: any;
}

export const WithReduxContext = (props: Props) => (
  <Provider store={store}>{props.children}</Provider>
);
