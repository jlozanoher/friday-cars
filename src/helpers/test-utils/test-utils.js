import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
import { makesReducer, modelsReducer, vehiclesReducer } from "../../slices";

function render(
  ui,
  {
    preloadedState,
    store = configureStore({
      reducer: {
        makes: makesReducer,
        models: modelsReducer,
        vehicles: vehiclesReducer,
      },
      preloadedState,
    }),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything
export * from "@testing-library/react";
// override render method
export { render };
