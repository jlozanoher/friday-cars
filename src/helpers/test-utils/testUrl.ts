export const testUrl = (path: string) => {
  return new URL(path, process.env.REACT_APP_SERVER_URL).toString();
};
