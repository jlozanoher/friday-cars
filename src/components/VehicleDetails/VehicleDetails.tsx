import cs from "classnames";
import React from "react";
import { useDispatch } from "react-redux";
import Vehicle from "../../models/vehicle.model";
import { setVehicleSelected } from "../../slices";
import { HStack, VStack } from "../Commons/Containers/Containers";
import { Center } from "../Commons/Result/styles";
import CarIcon from "./car-icon.png";
import * as S from "./styles";

interface Props {
  vehicle: Vehicle;
  selected?: boolean;
}

const VehicleDetails = (props: Props) => {
  const { vehicle: v, selected } = props;

  const dispatch = useDispatch();

  const handleSelect = () => {
    const r = !selected ? v : undefined;
    dispatch(setVehicleSelected(r));
  };

  return (
    <div data-testid={`VehicleDetails${v.id}`}>
      <S.Img src={CarIcon} alt={"car-icon"} />
      <HStack>
        <S.Title>{v.make}</S.Title>
        <S.SubTitle>{v.model}</S.SubTitle>
      </HStack>
      <VStack>
        <div>
          <small>Body type: {v.bodyType}</small>
        </div>
        <div>
          <small>Engine capacity: {v.engineCapacity} cc</small>
        </div>
        <div>
          <small>Power (Horsepower): {v.enginePowerPS}</small>
        </div>
        <div>
          <small>Power (KiloWatts): {v.enginePowerPW || "-"}</small>
        </div>
        <div>
          <small>Fuel type: {v.fuelType}</small>
        </div>
      </VStack>
      <div style={{ height: "20px" }} />
      <Center>
        <S.SelectBtn className={cs({ red: selected })} onClick={handleSelect}>
          {selected ? "Unselect" : "Select"}
        </S.SelectBtn>
      </Center>
    </div>
  );
};

export default VehicleDetails;
