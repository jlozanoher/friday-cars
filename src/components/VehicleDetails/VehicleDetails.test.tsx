import renderer from "react-test-renderer";
import { WithReduxContext } from "../../helpers/test-utils/WithReduxContext";
import Vehicle from "../../models/vehicle.model";
import VehicleDetails from "./VehicleDetails";

const vehicle: Vehicle = {
  id: "1",
  make: "BMW",
  model: "3er",
  bodyType: "Kombi",
  engineCapacity: 1995,
  enginePowerPS: 190,
  enginePowerPW: 300,
  fuelType: "Diesel",
};

describe("VehicleDetails should match snap", () => {
  it("VehicleDetails normal", () => {
    const tree = renderer
      .create(
        <WithReduxContext>
          <VehicleDetails vehicle={vehicle} />
        </WithReduxContext>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("VehicleDetails selected", () => {
    const tree = renderer
      .create(
        <WithReduxContext>
          <VehicleDetails vehicle={vehicle} selected />
        </WithReduxContext>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
