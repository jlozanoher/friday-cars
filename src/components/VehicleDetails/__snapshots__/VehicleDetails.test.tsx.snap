// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VehicleDetails should match snap VehicleDetails normal 1`] = `
<div
  data-testid="VehicleDetails1"
>
  <img
    alt="car-icon"
    className="sc-jRQBWg cILbES"
    src="car-icon.png"
  />
  <div
    className="sc-bdvvtL iuoSCa"
  >
    <div
      className="sc-hKwDye dMJKKO"
    >
      BMW
    </div>
    <div
      className="sc-eCImPb impPDT"
    >
      3er
    </div>
  </div>
  <div
    className="sc-gsDKAQ kkVWFs"
  >
    <div>
      <small>
        Body type: 
        Kombi
      </small>
    </div>
    <div>
      <small>
        Engine capacity: 
        1995
         cc
      </small>
    </div>
    <div>
      <small>
        Power (Horsepower): 
        190
      </small>
    </div>
    <div>
      <small>
        Power (KiloWatts): 
        300
      </small>
    </div>
    <div>
      <small>
        Fuel type: 
        Diesel
      </small>
    </div>
  </div>
  <div
    style={
      Object {
        "height": "20px",
      }
    }
  />
  <div
    className="sc-dkPtRN eZSDIs"
  >
    <div
      className="sc-gKclnd dqrdbM"
      onClick={[Function]}
    >
      Select
    </div>
  </div>
</div>
`;

exports[`VehicleDetails should match snap VehicleDetails selected 1`] = `
<div
  data-testid="VehicleDetails1"
>
  <img
    alt="car-icon"
    className="sc-jRQBWg cILbES"
    src="car-icon.png"
  />
  <div
    className="sc-bdvvtL iuoSCa"
  >
    <div
      className="sc-hKwDye dMJKKO"
    >
      BMW
    </div>
    <div
      className="sc-eCImPb impPDT"
    >
      3er
    </div>
  </div>
  <div
    className="sc-gsDKAQ kkVWFs"
  >
    <div>
      <small>
        Body type: 
        Kombi
      </small>
    </div>
    <div>
      <small>
        Engine capacity: 
        1995
         cc
      </small>
    </div>
    <div>
      <small>
        Power (Horsepower): 
        190
      </small>
    </div>
    <div>
      <small>
        Power (KiloWatts): 
        300
      </small>
    </div>
    <div>
      <small>
        Fuel type: 
        Diesel
      </small>
    </div>
  </div>
  <div
    style={
      Object {
        "height": "20px",
      }
    }
  />
  <div
    className="sc-dkPtRN eZSDIs"
  >
    <div
      className="sc-gKclnd dqrdbM red"
      onClick={[Function]}
    >
      Unselect
    </div>
  </div>
</div>
`;
